services/logging.cpp

```diff
@@ -63,7 +64,7 @@ void setup_debug(int level, const string &filename, const string &prefix)
 #ifdef __linux__
 
         if (fname[0] != '/') {
-            char buf[FILENAME_MAX];
+            char buf[PATH_MAX];
```

Comment:
After looking up PATH_MAX and FILENAME_MAX: both are incorrect.  linux imposes no limit on path lenghts and thus an exact constant cannot be defined.  The only correct thing a buffer (with malloc) and keep looping on getcwd with larger buffers until it succeeds.

FILENAME_MAX is too large for the stack in general though, so we shouldn't use it. (since we are single threaded I don't think this objection applies to us)